cmake/external_mlir.cmake

```diff
@@ -47,10 +47,23 @@ execute_process(COMMAND "${CMAKE_COMMAND}" --build . --target ext_mlir_llvm
 execute_process(COMMAND "${CMAKE_COMMAND}" --build . --target ext_mlir
                 WORKING_DIRECTORY "${MLIR_PROJECT_ROOT}")
 
-# point find_package to the pre-built libs
-set(LLVM_DIR ${MLIR_LLVM_ROOT}/build/lib/cmake/llvm)
+# Enable modules for LLVM.
+set(LLVM_DIR "${MLIR_BUILD_DIR}/lib/cmake/llvm"
+    CACHE PATH "Path to LLVM cmake modules")
+list(APPEND CMAKE_MODULE_PATH "${LLVM_DIR}")
```

Comment:
This is necessary ? My understanding is `LLVM_DIR` will direct `find_package` to right location of `LLVM` lib

Reply:
This is needed by the `include(AddLLVM)` below. I tried moving it after `find_package` to see if cmake module path was adjusted accordingly but it didn't work.